src/UiModes.ts

```diff
@@ -0,0 +1,321 @@
+        if(uiModeName === null){
+            UiModeSystem.setActiveUiMode(UiModes[2])
+        }else{
+            UiModeSystem.setActiveUiModeByName(uiModeName)
+        }
+    }
+
+    static saveToLocalStorage = () : void => {
+        if(this.localStorageUpdateCooldown===false){
+            this.localStorageUpdateCooldown = true;
+
+            setTimeout(function () {
+                const uiModesObj : any[] = []
+                UiModeSystem.getUiModes().forEach(function(uiMode:UiMode){
+                    const uiModeObj = {
+                        name : uiMode.getName(),
+                        description : uiMode.getDescription(),
+                        settingValues : <any[]>[]
+                    }
+                    uiMode.getSettings().forEach(function(setting:SettingData){
+                        const settingObj = {
+                            key : setting.getKey(),
+                            value : setting.getValue()
+                        }
+                        uiModeObj.settingValues.push(settingObj)
+                    })
+                    uiModesObj.push(uiModeObj)
+                })
+
+                localStorage.setItem('UiModes', JSON.stringify(uiModesObj));
+                localStorage.setItem('activeUiMode', UiModeSystem.getActiveUiMode().getName());
+                UiModeSystem.localStorageUpdateCooldown = false;
+            }, 1000)
+        }else{
+            return
+        }
+    }
+
+    static loadFromLocalStorage = () : void => {
+        const uiModesObj : any[] = JSON.parse(localStorage.getItem('UiModes'))
+
+        if(uiModesObj === null){
+            return
+        }
+
+        uiModesObj.forEach(function(uiModeObj){
+            let destUiMode : UiMode = UiModeSystem.getUiModeByName(uiModeObj.name)
+            if(destUiMode===null){
+                const settings : SettingData[] = []
+                
+                Setting.getSettings().forEach(function(settingsGroup){
+                    settingsGroup.getSettings().forEach(function(setting){
+                        settings.push(new SettingData(setting.getKey(),setting.getGraphDefaultVal(),setting.getPerpetual()))
+                    })
+                })
+
+                destUiMode = new UiMode(uiModeObj.name, uiModeObj.description, settings, false)
+            }
+            uiModeObj.settingValues.forEach(function(settingObj:any){
+                destUiMode.setSettingByKey(settingObj.key,settingObj.value)
+            } )
+
+        })
+    }
+
+    static updateSettingsArray = () : void => {
+        UiModeSystem.getActiveUiMode().getSettings().forEach(function(setting){
+            Setting.setValue(setting.getKey(),setting.getValue())
+        })
+    }
+
+    static getActiveSetting = (settingName:string) : any => {
+        let found = false
+        let value:any = null
+        UiModeSystem.getActiveUiMode().getSettings().forEach(function(setting){
```

Comment:
Just a matter of style/taste; but I prefer a for loop here rather than a forEach, since if can be terminated early when the setting with the correct key is found.

Maybe the input argument to this function should be "settingKey" or just "key" instead of "settingName", since you are matching it against setting.getKey()